test/indexed_test.dart

```diff
@@ -0,0 +1,199 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:math';
+import 'dart:typed_data';
+
+import 'package:collection/collection.dart';
+import 'package:test/test.dart';
+
+import '../lib/realm.dart';
+
+part 'indexed_test.g.dart';
+
+// IMPORTANT: Don't import our own test.dart here. It will break AOT compilation!
+
+// Realm models cannot extend other classes, but they can implement interfaces.
+class Base {
```

Comment:
Why do we need this here. If we want to test this out it seems we need to make this test part of `realm_object.dart`. Here it seems irrelevant since its never used.

Reply:
We no longer do. This is some residue from an earlier approach. I will remove `Base`

Reply:
Done